src/keyboard.c

```diff
@@ -45,6 +45,40 @@ struct keyboard* keyboard_new(struct wl_keyboard* wl_keyboard,
 	return self;
 }
 
+uint32_t* keyboard_get_pressed_key(struct keyboard* keyboard, uint32_t key)
+{
+	for (int i = 0; i < 256; i += 1) {
+		if (key == keyboard->pressed_keys[i]) {
+			return &keyboard->pressed_keys[i];
+		};
+	}
+	return NULL;
+}
+
+void keyboard_add_pressed_key(struct keyboard* keyboard, uint32_t key)
+{
+	if (keyboard_get_pressed_key(keyboard, key)) {
+		fprintf(stderr, "Oops, already pressed key.\n");
+		return;
+	}
+	uint32_t* pressed_key = keyboard_get_pressed_key(keyboard, 0);
+	if (NULL) {
+		fprintf(stderr, "Oops, pressed too many keys.\n");
+		return;
+	}
+	*pressed_key = key;
+}
+
+void keyboard_remove_pressed_key(struct keyboard* keyboard, uint32_t key)
+{
+	uint32_t* pressed_key = keyboard_get_pressed_key(keyboard, key);
```

Comment:
The algorithm used by wayvnc and labwc is a bit more elegant, but I guess this works too

Reply:
Agreed, but not doable without storing a size. There shouldn't be lot of pressed keys at the same time anyway